### PR TITLE
fix(auth): handle edge case

### PR DESCRIPTION
**Pull request by rudyardrichter**
- Export `get_public_key_for_kid` (needed in fence)
- Fix rogue `KeyError` which happens in `validate_request_jwt` if the header is malformed

### CODE REVIEW

File: cdispyutils/auth/jwt_validation.py [hunk 1]
```diff
@@ -173,6 +173,8 @@ def validate_request_jwt(
     request = request or flask.request
     try:
         encoded_token = request.headers['Authorization'].split(' ')[1]
+    except IndexError:
```
Comment (jbarno) [inline cdispyutils/auth/jwt_validation.py]: Catch all exceptions when parsing auth headers? Or does validate_jwt do that?

Comment (jbarno) [reply cdispyutils/auth/jwt_validation.py]: Also when you you say the word edge case and don't add a test case : http://i0.kym-cdn.com/photos/images/facebook/000/956/638/5bc.gif

Comment (rudyardrichter) [reply cdispyutils/auth/jwt_validation.py]: That statement specifically should only be able to raise `IndexError` or `KeyError` as far as I can tell. `validate_jwt` handles everything else, basically.

Thanks for pointing that out! I added a regression test for this, but in fence, not here…will add.

Comment (rudyardrichter) [reply cdispyutils/auth/jwt_validation.py]: Test case added in 866304c.
